lang/cs/Org.Apache.REEF.Driver/Bridge/Events/EvaluatorRequestor.cs

```diff
@@ -63,13 +67,24 @@ internal EvaluatorRequestor(IEvaluatorRequestorClr2Java clr2Java)
 
         public void Submit(IEvaluatorRequest request)
         {
-            LOGGER.Log(Level.Info, "Submitting request for {0} evaluators and {1} MB memory and  {2} core to rack {3}.", request.Number, request.MemoryMegaBytes, request.VirtualCore, request.Rack);
+            LOGGER.Log(Level.Info, "Submitting request for {0} evaluators and {1} MB memory and  {2} core to rack {3} and runtime {4}.", request.Number, request.MemoryMegaBytes, request.VirtualCore, request.Rack, request.RuntimeName);
 
+            byte[] data = Clr2Java.GetDefinedRuntimes();
+            DefinedRuntimes runtimes = DefinedRuntimesSerializer.FromBytes(data);
```

Comment:
If you look at the code in EvaluatorRequestorImpl.submit(),  all the validation are done in that class. So the Runtime validation should be there as well. Then you just need to inject the DefinedRuntimes in EvaluatorRequestorImpl  constructor and use it in the validation. No serialization is necessary for DefinedRuntimes.